thirdparty/faiss/faiss/cppcontrib/knowhere/IndexBruteForceWrapper.cpp

```diff
@@ -0,0 +1,123 @@
+// Copyright (C) 2019-2024 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License.
+
+#include <faiss/cppcontrib/knowhere/IndexBruteForceWrapper.h>
+
+#include <algorithm>
+#include <memory>
+
+#include <faiss/Index.h>
+#include <faiss/MetricType.h>
+#include <faiss/impl/AuxIndexStructures.h>
+#include <faiss/impl/DistanceComputer.h>
+#include <faiss/impl/FaissAssert.h>
+#include <faiss/impl/IDSelector.h>
+
+#include <faiss/cppcontrib/knowhere/impl/Bruteforce.h>
+
+namespace faiss {
+namespace cppcontrib {
+namespace knowhere {
+
+IndexBruteForceWrapper::IndexBruteForceWrapper(Index* underlying_index) :
```

Comment:
Why we have two BruteForce Wrappers? one in Knowhere and one in Faiss. And looks like nowhere call this one

Reply:
I expect to merge one, a generic one into Faiss, if they accept it. Another one is a specialized version for our needs. The other one might be removed in Faiss guys reject it.